lib/core/src/animation/easing.rs

```diff
@@ -0,0 +1,118 @@
+//! This file provides several easing functions used for transition animation.
```

Comment:
Everything here looks nice and cool! One thing raised my attention. Everything here is defined on `f64`. In many cases, I will be animating `f32` values. Casting them back and forth is rather a bad idea. Can we make these implementations polymorphic? At least for both `f64` and `f32`? If not (If that would be hard), then I would prefer to choose `f32` over `f64`, as all WebGL values are `f32`.

Reply:
I made it f32. I tried to make polymorphic using nalgebra::RealField, but it wasn't trivial. I can work on it later.

I added this comment

```rust
// TODO: It would be useful to make the easing functions polymorphic. Maybe we can make use of
// nalgebra::RealField?
```